src/interpreter/mod.rs

```diff
@@ -143,18 +144,36 @@ struct ConstantId<'tcx> {
     kind: ConstantKind,
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 enum ConstantKind {
     Promoted(mir::Promoted),
     /// Statics, constants and associated constants
     Global,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub struct Constant<'tcx> {
+    data: Option<Value>,
+    mutable: bool,
```

Comment:
@eddyb Global might be a better name, but in Miri even constants are mutable while we are initializing them.